galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistUtils.java

```diff
@@ -0,0 +1,214 @@
+import dev.galasa.framework.spi.IDynamicStatusStoreService;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IResultArchiveStoreDirectoryService;
+import dev.galasa.framework.spi.IRunResult;
+import dev.galasa.framework.spi.ResultArchiveStoreException;
+
+public class WorklistUtils {
+	
+	private IFramework framework;
+	
+	public WorklistUtils(IFramework framework) {
+		this.framework = framework;
+	}
+	
+	
+	protected List<String> getRunIdsFromDss() throws DynamicStatusStoreException, ServletException {
+	
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		String worklistProperty = dss.get("user.username.worklist");
+		
+		if (worklistProperty != null) {
+			
+			List<String> runIdsInWorklist = new ArrayList<String>();	
+			
+			if (worklistProperty.length() != 0) {
+				
+				JsonObject jsonObject = new JsonParser().parse(worklistProperty).getAsJsonObject();
+				
+				JsonArray runIds = (JsonArray) jsonObject.get("runIds");		
+
+				for (JsonElement el : runIds) {
+					runIdsInWorklist.add(el.getAsString());
+				}	
+			}	
+			
+			return runIdsInWorklist;
+			
+		} else {
+			return null;
+		}
+		
+	}
+	
+	
+	protected List<WorklistItem> getWorklistItems() throws ServletException {
+		
+		List<WorklistItem> runs = new ArrayList<WorklistItem>();
+		
+		if (WorklistQuery.runIdsInWorklist != null) {
+			
+			IRunResult run = null;
+			for (String runId : WorklistQuery.runIdsInWorklist) {
+				try {
+					run = getRunById(runId);
+				} catch (ResultArchiveStoreException e) {
+					throw new ServletException("Error while retrieving run for Run ID " + runId, e);
+				}
+				
+				if (run != null) {
+					
+					try {
+						
+						String runName = run.getTestStructure().getRunName();
+						String shortName = run.getTestStructure().getTestShortName();
+						String testClass = run.getTestStructure().getTestName();
+						String result = run.getTestStructure().getResult();
+						
+						WorklistItem worklistItem = new WorklistItem(runId, runName, shortName, testClass, result);
+						
+						runs.add(worklistItem);
+						
+					} catch (ResultArchiveStoreException e) {
+						throw new ServletException("Error while retriving Worklist information for Run ID " + runId, e);
+					}
+					
+				} else {
+					WorklistQuery.runIdsNotFound.add(runId);
+					try {
+						removeRunIdFromDss(runId);
+					} catch (Exception e) {
+						throw new ServletException("Error while updating the DSS", e);
+					}
+				}
+				
+			}
+		}
+		
+		return runs;
+	}
+	
+	protected IRunResult getRunById(String id) throws ResultArchiveStoreException {
+
+		IRunResult run = null;
+	       
+        for (IResultArchiveStoreDirectoryService directoryService : framework.getResultArchiveStore().getDirectoryServices()) {
+           
+           run = directoryService.getRunById(id);
+           
+           if(run != null) {
+              break;
+           }
+           
+        }
+		
+		return run;
```

Comment:
if you do return run; above,  then this can be return null;

it is clearer what is happening